contracts/0.4.24/Lido.sol

```diff
@@ -355,342 +463,414 @@ contract Lido is ILido, StETH, AragonApp {
+     */
+    function _processClStateUpdate(
+        uint256 _preClValidators,
+        uint256 _postClValidators,
+        uint256 _postClBalance
+    ) internal returns (uint256 preCLBalance) {
+        uint256 depositedValidators = DEPOSITED_VALIDATORS_POSITION.getStorageUint256();
+        require(_postClValidators <= depositedValidators, "REPORTED_MORE_DEPOSITED");
 
-    /**
-    * @notice Get the amount of Ether temporary buffered on this contract balance
-    * @dev Buffered balance is kept on the contract from the moment the funds are received from user
-    * until the moment they are actually sent to the official Deposit contract.
-    * @return amount of buffered funds in wei
-    */
-    function getBufferedEther() external view returns (uint256) {
-        return _getBufferedEther();
-    }
+        require(_postClValidators >= _preClValidators, "REPORTED_LESS_VALIDATORS");
 
-    /**
-    * @notice Get total amount of execution layer rewards collected to Lido contract
-    * @dev Ether got through LidoExecutionLayerRewardsVault is kept on this contract's balance the same way
-    * as other buffered Ether is kept (until it gets deposited)
-    * @return amount of funds received as execution layer rewards (in wei)
-    */
-    function getTotalELRewardsCollected() external view returns (uint256) {
-        return TOTAL_EL_REWARDS_COLLECTED_POSITION.getStorageUint256();
-    }
 
-    /**
-    * @notice Get limit in basis points to amount of ETH to withdraw per LidoOracle report
-    * @return limit in basis points to amount of ETH to withdraw per LidoOracle report
-    */
-    function getELRewardsWithdrawalLimit() external view returns (uint256) {
-        return EL_REWARDS_WITHDRAWAL_LIMIT_POSITION.getStorageUint256();
-    }
+        if (_postClValidators > _preClValidators) {
+            CL_VALIDATORS_POSITION.setStorageUint256(_postClValidators);
+        }
 
-    /**
-    * @notice Gets deposit contract handle
-    */
-    function getDepositContract() public view returns (IDepositContract) {
-        return IDepositContract(DEPOSIT_CONTRACT_POSITION.getStorageAddress());
-    }
+        uint256 appearedValidators = _postClValidators.sub(_preClValidators);
+        preCLBalance = CL_BALANCE_POSITION.getStorageUint256();
+        // Take into account the balance of the newly appeared validators
+        preCLBalance = preCLBalance.add(appearedValidators.mul(DEPOSIT_SIZE));
 
-    /**
-    * @notice Gets authorized oracle address
-    * @return address of oracle contract
-    */
-    function getOracle() public view returns (address) {
-        return ORACLE_POSITION.getStorageAddress();
+        // Save the current CL balance and validators to
+        // calculate rewards on the next push
+        CL_BALANCE_POSITION.setStorageUint256(_postClBalance);
     }
 
     /**
-    * @notice Gets node operators registry interface handle
-    */
-    function getOperators() public view returns (INodeOperatorsRegistry) {
```

Comment:
😢

Reply:
???

Reply:
Should we really leave it here?

Reply:
not sure where it might break, a lot of tooling is tied to this, we need to check everything

Reply:
adding back then

Reply:
Looks like it's better to remove this one.
We don't have any sources for the NOR address from Lido or LidoLocator.